### DISCUSSION TITLE
Debug Frontend

### DISCUSSION THREAD

**Discussion by John0x** [tauri-apps/tauri#5227] [Q&A]
Hey, is there a way to debug the frontend (vite with reactjs in my case) from vs code?
I know that you can open the devtools, which should probably mean that you can debug it remotely too, right?

But it doesn't show up as a target for remote debugging (edge/chrome devtools)

**Comment by FabianLars** [comment]
Does this issue help you? https://github.com/tauri-apps/tauri-docs/issues/842

**Reply by John0x** [reply]
> Does this issue help you? [tauri-apps/tauri-docs#842](https://github.com/tauri-apps/tauri-docs/issues/842)

I'm on macos unfortunately. The linked issue seems to have a solution for windows only :/

**Reply by FabianLars** [reply]
to our knowledge wkwebview (the webkit based webview on macos) doesn't support this kind of stuff, same as webkitgtk on linux.

If your frontend doesn't call any Tauri apis or can put them behind an if statement (unless they are the things you wanna debug), you could run your frontend in a normal browser or debugging environment. For most Tauri APIs (basically all except window manipulation) you can use tauri-invoke-http as mentioned in the issue to connect the browser instance to the Tauri backend.

Other than this, actual frontend debugging on macos/linux is not reallyyy something we can offer with the tools we have at hand :/

**Reply by John0x** [reply]
Is there maybe a way to opt-in to use chromium instead of wkwebview for development on mac? Being able to debug the frontend is a pretty big part of the development cycle.
Unfortunately, the app relies heavily on the native parts, provided by custom tauri commands.

**Reply by FabianLars** [reply]
Since we can't use anything based on chromium on macos (because there is no usable wkwebview alternative), you need to use your browser somehow. This was the primary reason why tauri-invoke-http and tauri-plugin-localhost were created.

**Reply by John0x** [reply]
Okay, how would that work?
I don't really see an example on how to integrate it with your frontend. Is there a different invoke command or do I have to include the local address in the invoke call?

**Reply by FabianLars** [reply]
Ideally you don't have to do anything. It's supposed to be plug and play apart from the small rust code changes

**Reply by John0x** [reply]
I've installed tauri-invoke-http and set it up, like mentioned in the Readme. It works fine if running within the tauri window but how would the browser know about it?
My frontend is running in a vite dev server, which is running independently from tauri. How would the javascript invoke function know where to send the request to? Am I missing something?

**Reply by FabianLars** [reply]
Okay yeah, i think i misremembered how invoke-http works _again_ lol. So you indeed need to use the localhost plugin, which of course breaks vite :(

I guess we need to revisit this topic in the near future, because this is indeed quite an unfortunate situation we have here...

**Reply by John0x** [reply]
Thank you for your effort :)

This should indeed be revisited. Debugging is a pretty big part of the development cycle, and not having a debugger would make that pretty annoying 👀
